Jenkinsfile

```diff
@@ -335,6 +356,99 @@ az account set -s $AZURE_SUBSCRIPTION_ID
+ --project ${project} \
+ --create-subnetwork '' \
+ --enable-ip-alias \
+ --num-nodes 3 \
+ --zone ${zone} \
+ --enable-autorepair \
+ --labels 'platform=${gcpLabel(platform)},branch=${gcpLabel(BRANCH_NAME)},build=${gcpLabel(BUILD_TAG)}'
+"""
+                                                sh "gcloud container clusters get-credentials ${clusterName} --zone ${zone} --project ${project}"
+                                                sh "kubectl create clusterrolebinding cluster-admin-binding --clusterrole=cluster-admin --user=\$(gcloud info --format='value(config.account)')"
+
+                                                waitForRollout("kube-system", 30)
+                                            }
+
+                                            // Reuse this service principal for externalDNS and oauth2.  A real (paranoid) production setup would use separate minimal service principals here.
+                                            def saCreds = JsonOutput.toJson(readFile(env.GOOGLE_APPLICATION_CREDENTIALS))
+
+                                            // NB: writeJSON doesn't work without approvals(?)
+                                            // See https://issues.jenkins-ci.org/browse/JENKINS-44587
+                                            writeFile([file: 'kubeprod-autogen.json', text: """
+{
+  "dnsZone": "${dnsZone}",
+  "externalDns": {
+    "credentials": ${saCreds}
+  },
+  "oauthProxy": {
+    "client_id": "${OAUTH_CLIENT_ID}",
+    "client_secret": "${OAUTH_CLIENT_SECRET}"
+  }
+}
+"""
+                                            ])
+
+
+                                            writeFile([file: 'kubeprod-manifest.jsonnet', text: """
+(import "manifests/platforms/gke.jsonnet") {
+  config:: import "kubeprod-autogen.json",
+  letsencrypt_environment: "staging",
+  prometheus+: import "tests/testdata/prometheus-crashloop-alerts.jsonnet",
+}
+"""
+                                            ])
+                                        }
+                                    }
+                                    finally {
+                                        container('gcloud') {
+                                            def disksFilter = "${clusterName}".take(18).replaceAll(/-$/, '')
+                                            sh "gcloud auth activate-service-account --key-file ${GOOGLE_APPLICATION_CREDENTIALS}"
+                                            sh "gcloud container clusters delete ${clusterName} --zone ${zone} --project ${project} --quiet || :"
+                                            sh "gcloud compute disks delete \$(gcloud compute disks list --project ${project} --filter name:${disksFilter} --format='value(name)') --project ${project} --zone ${zone} --quiet || :"
+                                            sh "gcloud dns managed-zones delete \$(gcloud dns managed-zones list --filter dnsName:${dnsZone} --format='value(name)' --project ${project}) --project ${project} || :"
```

Comment:
Aside: If it becomes relevant at some point, we can easily delete all the entries in a managed zone using:
```
gcloud dns record-sets import /dev/null --zone=$zone --delete-all-existing
```

Reply:
thanks for this. Using this now and reverted back to deleting the cluster asynchronously.